server/services/prisonerService.ts

```diff
@@ -66,6 +66,22 @@ export default class PrisonerService {
     return this.prisonerSearchApiClient.searchPrisonersByBookingIds(prisonerSearchCriteria, user)
   }
 
+  async getActiveHdcStatus(bookingId: string): Promise<HdcStatus | null> {
+    const hdcLicence = await this.prisonApiClient.getLatestHdcStatus(bookingId)
+    if (!hdcLicence) return null
+
+    const hdcBookingId = hdcLicence.bookingId.toString()
+    if (hdcBookingId !== bookingId) {
+      throw new Error('BookingId for HDC licence does not match submitted BookingId')
```

Comment:
Would this ever not match? Its the booking ID that we called the function with, so should always match?

Reply:
Yes you are right. Its very defensive programming. The only scenario where this could be a problem is when a bug is introduced at a later point in the other service. I can remove this one.